use-cases.md

```diff
@@ -53,11 +53,7 @@ When a server wants to trace execution of a request, it generally needs to go th
 Let's assume that we have an HTTP server, and the Span is propagated from the client via HTTP headers, accessible via `request.headers`:
 
 ```python
-    span = tracer.join_trace_from_text(
-        operation_name=operation,
-        context_snapshot=request.headers,
-        trace_attributes=request.headers
-    )
+    span = tracer.get_span_extractor(request.class).extract(request)
```

Comment:
@michaelsembwever I misunderstood your proposal, I thought your `request.class` was something like `HttpRequest.class`.  If it's instead a basic data type line byte[] or map, then it's certainly doable (should be made explicit in the doc though). And the `request.class` can even be one of the streaming-oriented APIs I mentioned above, +1 for that. Another thing I like is that it abstracts away the separation of spanID + traceAttrs to a certain degree (although it's still a question how to do it in the binary form of this approach).

The main concern people will have with that is the same one as on the `setTag(String, Object)` signature. There's no guarantee that the end user won't supply some nonsense as an argument, and the tracer will not be able to propagate the context. The spec for the tracer would be requiring to support a certain minimal set of `request.class`es, and if it supports more, end users can't really rely on that in their code. So flexibility in this regard hurts portability.